chia/wallet/dao_wallet/dao_wallet.py

```diff
@@ -0,0 +1,2268 @@
+            to_puzzle_hash=dao_treasury_puzzle.get_tree_hash(),
+            amount=uint64(1),
+            fee_amount=fee,
+            confirmed=False,
+            sent=uint32(10),
+            spend_bundle=full_spend,
+            additions=full_spend.additions(),
+            removals=full_spend.removals(),
+            wallet_id=self.id(),
+            sent_to=[],
+            trade_id=None,
+            type=uint32(TransactionType.INCOMING_TX.value),
+            name=bytes32(token_bytes()),
+            memos=[],
+        )
+        regular_record = dataclasses.replace(tx_record, spend_bundle=None)
+        await self.wallet_state_manager.add_pending_transaction(regular_record)
+        await self.wallet_state_manager.add_pending_transaction(treasury_record)
+        await self.wallet_state_manager.add_interested_puzzle_hashes([launcher_coin.name()], [self.id()])
+        await self.wallet_state_manager.add_interested_coin_ids([launcher_coin.name()], [self.wallet_id])
+
+        await self.wallet_state_manager.add_interested_coin_ids([eve_coin.name()], [self.wallet_id])
+        return full_spend
+
+    async def generate_treasury_eve_spend(
+        self, inner_puz: Program, eve_coin: Coin, fee: uint64 = uint64(0)
+    ) -> SpendBundle:
+        """
+        Create the eve spend of the treasury
+        This can only be completed after a number of blocks > oracle_spend_delay have been farmed
+        """
+        if self.dao_info.current_treasury_innerpuz is None:  # pragma: no cover
+            raise ValueError("generate_treasury_eve_spend called with nil self.dao_info.current_treasury_innerpuz")
+        full_treasury_puzzle = curry_singleton(self.dao_info.treasury_id, inner_puz)
+        launcher_id, launcher_proof = self.dao_info.parent_info[0]
+        assert launcher_proof
+        assert inner_puz
+        inner_sol = Program.to([0, 0, 0, 0, get_singleton_struct_for_id(launcher_id)])
+        fullsol = Program.to(
+            [
+                launcher_proof.to_program(),
+                eve_coin.amount,
+                inner_sol,
+            ]
+        )
+        eve_coin_spend = CoinSpend(eve_coin, full_treasury_puzzle, fullsol)
+        eve_spend_bundle = SpendBundle([eve_coin_spend], G2Element())
+
+        next_proof = LineageProof(
+            eve_coin.parent_coin_info,
+            inner_puz.get_tree_hash(),
+            uint64(eve_coin.amount),
+        )
+        next_coin = Coin(eve_coin.name(), eve_coin.puzzle_hash, eve_coin.amount)
+        await self.add_parent(eve_coin.name(), next_proof)
+        await self.wallet_state_manager.add_interested_coin_ids([next_coin.name()], [self.wallet_id])
+
+        dao_info = dataclasses.replace(self.dao_info, current_treasury_coin=next_coin)
+        await self.save_info(dao_info)
+        await self.wallet_state_manager.singleton_store.add_spend(self.id(), eve_coin_spend)
+        return eve_spend_bundle
+
+    def generate_simple_proposal_innerpuz(
```

Comment:
These `generate_` functions seem like maybe they belong in the dao_utils file

Reply:
moved to utils